newsroom/users/views.py

```diff
@@ -80,22 +80,26 @@ def user_profile():
 def search():
     lookup = None
     sort = None
-    if flask.request.args.get("q"):
-        regex = re.compile(".*{}.*".format(flask.request.args.get("q")), re.IGNORECASE)
-        lookup = {"$or": [{"first_name": regex}, {"last_name": regex}]}
-
-    if flask.request.args.get("ids"):
-        lookup = {"_id": {"$in": (flask.request.args.get("ids") or "").split(",")}}
-
-    if flask.request.args.get("sort"):
-        sort = flask.request.args.get("sort")
-
-    if flask.request.args.get("where"):
-        where = json.loads(flask.request.args.get("where"))
-        if where.get("company"):
-            lookup = {"company": where.get("company")}
-        if where.get("products._id"):
-            lookup = {"products._id": where.get("products._id")}
+    try:
+        if flask.request.args.get("q"):
+            regex = re.compile(".*{}.*".format(flask.request.args.get("q")), re.IGNORECASE)
+            lookup = {"$or": [{"first_name": regex}, {"last_name": regex}]}
+
+        if flask.request.args.get("ids"):
+            lookup = {"_id": {"$in": (flask.request.args.get("ids") or "").split(",")}}
+
+        if flask.request.args.get("sort"):
+            sort = flask.request.args.get("sort")
+
+        where_param = flask.request.args.get("where")
+        if where_param:
+            where = json.loads(where_param)
+            if where.get("company"):
+                lookup = {"company": where["company"]}
+            if where.get("products._id"):
+                lookup = {"products._id": where["products._id"]}
+    except json.JSONDecodeError as e:
```

Comment:
would actually put it that `try/catch` only around `json.loads` we don't need it for the whole thing